docs/src/architecture/accounts.md

```diff
@@ -13,63 +16,56 @@ In the above:
+| | Regular updatable account | Regular immutable account | Faucet for fungible assets | Faucet for non-fungible assets |
+|---|---|---|---|---|
+| **Description** | Will be used by most users for a wallet. Code specification and changes possible. | Will be used by most smart contracts. Once deployed code cannot be changed | Users can issue fungible assets and customize them. | Users can issue non-fungible assets and customize them. |
+| **Code updatability** | yes | no | no | no |
+| **Most significant bits** | `00` | `01` | `10` | `11` |
 
-### Example account ID
+#### Example account ID (big-endian)
 
 <p align="center">
     <img src="../diagrams/architecture/account/Account_ID.png">
 </p>
 
-## Storage
-User-defined data that can be stored in an account. Account storage is composed of two components. The first component is a simple sparse Merkle tree of depth 8 which is index addressable. This provides the user with 256 Word slots. Users that require additional storage can use the second component which is a `MerkleStore`. This will allow the user to store any Merkle structures they need. This is achieved by storing the root of the Merkle structure as a leaf in the simple sparse merkle tree. When `AccountStorage` is serialized it will check to see if any of the leafs in the simple sparse Merkle tree are Merkle roots of other Merkle structures. If any Merkle roots are found then the Merkle structures will be persisted in the `AccountStorage` `MerkleStore`.
+*Note: Miden uses little-endian by default, but big_endian for better readability in the picture above.*
 
-TODO ADD STORAGE DIAGRAM
+### Account Storage
+User-defined data that can be stored in an account. `AccountStorage` is composed of two components. 
 
-## Nonce
+The first component is a simple sparse Merkle tree of depth `8` which is index addressable. This provides the user with `256` `Word` slots. 
+
+Users requiring additional storage can use the second component a `MerkleStore`. It allows users to store any Merkle structures they need. The root of the Merkle structure can be stored as a leaf in the simple sparse Merkle tree. When `AccountStorage` is serialized it will check to see if any of the leafs in the simple sparse Merkle tree are Merkle roots of other Merkle structures. If any Merkle roots are found then the Merkle structures will be persisted in the `AccountStorage` `MerkleStore`.
+
+### Nonce
 Counter which must be incremented whenever the account state changes. Nonce values must be strictly monotonically increasing and can be incremented by any value smaller than 2^{32} for every account update.
 
-## Vault
+### Vault
 Asset container for an account.
 
-An account vault can contain an unlimited number of assets. The assets are stored in a Sparse
+An account vault can contain an unlimited number of [assets](https://0xpolygonmiden.github.io/miden-base/architecture/assets.html). The assets are stored in a sparse
```

Comment:
There is a technical limit - the number of assets that can be represented using our asset model.

Reply:
Is the limit the max number in our field? Or what is the technical limit?

Reply:
For fungible assets its the maximum number of unique faucet accounts that can be created which is ~2^58.  For non fungible assets it's the maximum number of unique faucet accounts combined with the max number of unique NFT's the account can mint.  I believe this is ~2^251.